apps/src/code-studio/pd/pre_workshop_survey/PreWorkshopQuestions.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import FormComponent from '../form_components/FormComponent';
+
+export default class PreWorkshopQuestions extends FormComponent {
+  handleChange(newState) {
+    // Clear the lesson any time a new unit is selected.
+    if ('unit' in newState) {
+      newState.lesson = null;
+    }
+    super.handleChange(newState);
```

Comment:
I don't _think_ `handleChange` is super-opinionated about state mutation, but this feels like it would be a lot more React-y if didn't actually mutate `newState`